core/src/services/sqlite/backend.rs

```diff
@@ -0,0 +1,257 @@
+    const SCHEME: Scheme = Scheme::Sqlite;
+    type Accessor = SqliteBackend;
+
+    fn from_map(map: HashMap<String, String>) -> Self {
+        let mut builder = SqliteBuilder::default();
+        map.get("connection_string")
+            .map(|v| builder.connection_string(v));
+        map.get("table").map(|v| builder.table(v));
+        map.get("key_field").map(|v| builder.key_field(v));
+        map.get("value_field").map(|v| builder.value_field(v));
+        map.get("root").map(|v| builder.root(v));
+        builder
+    }
+
+    fn build(&mut self) -> Result<Self::Accessor> {
+        let connection_string = match self.connection_string.clone() {
+            Some(v) => v,
+            None => {
+                return Err(Error::new(
+                    ErrorKind::ConfigInvalid,
+                    "connection_string is required but not set",
+                )
+                .with_context("service", Scheme::Sqlite))
+            }
+        };
+        let table = match self.table.clone() {
+            Some(v) => v,
+            None => {
+                return Err(Error::new(ErrorKind::ConfigInvalid, "table is empty")
+                    .with_context("service", Scheme::Postgresql))
+            }
+        };
+        let key_field = match self.key_field.clone() {
+            Some(v) => v,
+            None => "key".to_string(),
+        };
+        let value_field = match self.value_field.clone() {
+            Some(v) => v,
+            None => "value".to_string(),
+        };
+        let root = normalize_root(
+            self.root
+                .clone()
+                .unwrap_or_else(|| "/".to_string())
+                .as_str(),
+        );
+        Ok(SqliteBackend::new(Adapter {
+            connection_string: connection_string,
+            table: table,
+            key_field: key_field,
+            value_field: value_field,
+        })
+        .with_root(&root))
+    }
+}
+
+pub type SqliteBackend = kv::Backend<Adapter>;
+
+#[derive(Clone)]
+pub struct Adapter {
+    connection_string: String,
+
+    table: String,
+    key_field: String,
+    value_field: String,
+}
+
+impl Debug for Adapter {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut ds = f.debug_struct("SqliteAdapter");
+        ds.field("connection_string", &self.connection_string);
+        ds.field("table", &self.table);
+        ds.field("key_field", &self.key_field);
+        ds.field("value_field", &self.value_field);
+        ds.finish()
+    }
+}
+
+#[async_trait]
+impl kv::Adapter for Adapter {
+    fn metadata(&self) -> kv::Metadata {
+        kv::Metadata::new(
+            Scheme::Sqlite,
+            &self.table,
+            Capability {
+                read: true,
+                write: true,
+                create_dir: true,
+                delete: true,
+                ..Default::default()
+            },
+        )
+    }
+
+    async fn get(&self, path: &str) -> Result<Option<Vec<u8>>> {
```

Comment:
We can implement `blocking_get` and calling `tokio::task::spawn_blocking()` in async context.